### PR TITLE
[SAAS-8997] Refactor fargate deployment action

### PR DESCRIPTION
**Pull request by lisa-tw**

### CODE REVIEW

File: ecrDeployment.js [hunk 1]
```diff
@@ -48,25 +50,22 @@ async function buildAndPushMultiArchImage(
...
-  console.log('Starting to build docker images...');
+  console.log('Starting to build docker image for ARM architecture ...');
   const addGithubTokenCommand = getAddGithubTokenEnvCommand();
   await exec.exec('docker buildx create --use');
+
   console.log('Pushing to ECR...');
   const dockerFilename = shouldUseCustomDockerfile ? '-f Dockerfile.dev' : '';
-
   await exec.exec(
-    `docker buildx build ${addGithubTokenCommand} -t ${ecrRepo}:${version}-amd64 ${dockerFilename} --platform linux/amd64 . --push`
```
Comment (vladovsiychuk) [inline ecrDeployment.js:58]: Amd architecture build no longer needed?

File: ecrDeployment.js [hunk 2]
```diff
@@ -48,25 +50,22 @@ async function buildAndPushMultiArchImage(
...
-  await exec.exec(
-    `docker buildx build ${addGithubTokenCommand} -t ${ecrRepo}:${version}-arm64 ${dockerFilename} --platform linux/arm64 . --push`
-  );
-
   console.log('Successfully pushed to ECR');
-  console.log('Creating manifest list ...');
+
+  console.log('Creating manifest list for :latest tag...');
   await exec.exec(
-    `docker manifest create ${ecrRepo}:${version} ${ecrRepo}:${version}-amd64 ${ecrRepo}:${version}-arm64`
```
Comment (vladovsiychuk) [inline ecrDeployment.js:67]: If we no longer need the amd build then please rename the function from `buildAndPushMultiArchImage` to something like `buildAndPushArmImage`

Comment (lisa-tw) [reply ecrDeployment.js:67]: Renamed to `buildAndPushArmImageAndManifestList`

File: ecrDeployment.js (hunk 1 above)
Comment (lisa-tw) [reply ecrDeployment.js:58]: It is needed. I am calling the `buildAndPushImage` function always and the `buildAndPushArmImageAndManifestList` function only if the env is dev

Comment (vladovsiychuk) [reply ecrDeployment.js:58]: Does this mean that the image will not work correctly on a PC with an Intel processor?

Comment (vladovsiychuk) [reply ecrDeployment.js:58]: A yeah, I see

**Review by vladovsiychuk — APPROVED** [review]

**Pull request merged**